oauth2client/client.py

```diff
@@ -279,8 +280,7 @@ def new_from_json(cls, s):
       An instance of the subclass of Credentials that was serialized with
       to_json().
     """
-    if isinstance(s, bytes):
-      s = s.decode('utf-8')
+    s = _from_bytes(s)
```

Comment:
Might it be worth using a different name for the result of _from_bytes, since it is potentially of a different type?